### PR TITLE
Add task4 Collections

### PR DESCRIPTION
**Pull request by NadejdaClima**

### CODE REVIEW

File: src/main/java/com/endava/soa_5ed/task4/collections/UsersFilter.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.endava.soa_5ed.task4.collections;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class UsersFilter {
```
Comment (EugenS21) [inline src/main/java/com/endava/soa_5ed/task4/collections/UsersFilter.java:9]: If this is a kind o Utility Class, you may create a private constructor and make the methods static.

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class User {
+    protected final Map<String, Address> addresses;
+    private long id;
+    private String username;
```
Comment (EugenS21) [inline src/main/java/com/endava/soa_5ed/task4/collections/User.java]: You don't have any setter for this fields - so you may make them final too.

**Review by EugenS21 — CHANGES_REQUESTED** [review]

File: src/main/java/com/endava/soa_5ed/task4/collections/UsersFilter.java (hunk 1 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/UsersFilter.java:9]: Done

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java (hunk 2 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/User.java]: Done

**Review by margaritaq — APPROVED** [review]

File: src/main/java/com/endava/soa_5ed/task3_oop/Cat.java [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
 
     public Cat(String gender, String furColor, String name, double size) {
         super(gender, furColor);
         this.name = name;
         this.size = size;
+        log.info("Creating Cat with gender: " + gender + ", fur color: " + furColor + ", name: " + name + " and size: " + size);
     }
 
     public double getSize() {
+
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task3_oop/Cat.java]: remove empty line

File: src/main/java/com/endava/soa_5ed/task4/collections/Order.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package com.endava.soa_5ed.task4.collections;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class Order {
+    public List<Item> items;
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task4/collections/Order.java]: make it private please

File: src/main/java/com/endava/soa_5ed/task4/collections/Order.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    private final long id;
+    private final LocalDate date;
+    private User user;
+
+    public Order() {
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task4/collections/Order.java]: make is private please so no one could use it outside of this class

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
+package com.endava.soa_5ed.task4.collections;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class User {
+    protected final Map<String, Address> addresses;
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task4/collections/User.java]: make the attributes private please.

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    protected final long id;
+    protected String username;
+
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task4/collections/User.java:13]: remove 1 empty line

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    public User() {
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5ed/task4/collections/User.java]: make is private please

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java (hunk 7 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/User.java:13]: Done

File: src/main/java/com/endava/soa_5ed/task3_oop/Cat.java (hunk 3 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task3_oop/Cat.java]: Done

File: src/main/java/com/endava/soa_5ed/task4/collections/Order.java (hunk 4 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/Order.java]: Done

File: src/main/java/com/endava/soa_5ed/task4/collections/Order.java (hunk 5 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/Order.java]: Done

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java (hunk 6 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/User.java]: Done

File: src/main/java/com/endava/soa_5ed/task4/collections/User.java (hunk 8 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5ed/task4/collections/User.java]: Done

**Review by EugenS21 — APPROVED** [review]

**Review by MarcelMorozniuc — APPROVED** [review]

**Pull request merged**